tests/control/custom.test.js

```diff
@@ -25,10 +26,202 @@ describe('Test Custom Control', () => {
     const fb = await $(fbWrap).formBuilder({fields, templates}).promise
     const field = {
       type: 'starRating',
-      class: 'form-control'
+      className: 'form-control'
     }
     fb.actions.addField(field)
 
     expect(cbOnRender.mock.calls).toHaveLength(1)
+
+    $(fbWrap).find('li.input-control[data-type="starRating"]').click()
+
+    expect(cbOnRender.mock.calls).toHaveLength(2)
+  })
+
+  test('test rendering custom field with template', async () => {
+    const fbWrap = $('<div>')
+    const cbOnRender = jest.fn()
+
+    const formData = [
+      {
+        'type': 'starRating',
+        'required': false,
+        'label': 'Star Rating',
+        'name': 'starRating-1697591966052-0'
+      },
+    ]
+    const templates = {
+      starRating: function(fieldData) {
+        return {
+          field: '<span id="'+fieldData.name+'">',
+          onRender: cbOnRender
+        }
+      }
+    }
+
+    const fr = await $(fbWrap).formRender({formData, templates}).promise
+
+    expect(cbOnRender.mock.calls).toHaveLength(1)
+
+    expect($(fbWrap).find('#starRating-1697591966052-0')[0].outerHTML).toBe('<span id="starRating-1697591966052-0"></span>')
+
+  })
+
+  test('override built-in with template', async () => {
+    const fbWrap = $('<div>')
+    const cbOnRender = jest.fn()
+
+    const fields = [
+      {
+        type: 'checkbox-group',
+        subtype: 'custom-group',
+        label: 'Custom Checkbox Group w/Sub Type',
+        required: !0,
```

Comment:
just curious, why not just `true`?

Reply:
Can't answer that. I'm not sure if it was my IDE that did that or if I copy/pasted the example from somewhere. I know webpack has a habit of using !0 for true